core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -287,18 +287,20 @@ class CompactEntityQuery(driverComponent: DriverComponent)
                                  destWs: UUID,
                                  entityRefs: Set[EntityPointer] = Set(),
                                  batchSize: Int = driverComponent.batchSize
-  ): ReadWriteAction[Int] = {
+  ): WriteAction[(Int, Int)] = {
 
     def copyChunkOfEntitiesOrAllEntities(chunk: Set[EntityPointer] = Set()) =
       for {
         entitiesCopiedCount <- copyEntities(sourceWs, destWs, chunk)
       } yield entitiesCopiedCount
 
-    val chunks: Iterator[Set[EntityPointer]] = entityRefs.grouped(batchSize)
-
+    val chunks: Iterator[Set[EntityPointer]] = if (entityRefs.size > batchSize) {
+      entityRefs.grouped(batchSize)
+    } else {
+      Iterator(entityRefs)
+    }
```

Comment:
I don't think you need this change - you can just call `entityRefs.grouped(batchSize)` and it will handle the if condition internally.